Flaky test: test_cholesky

`tests/chainer_tests/functions_tests/math_tests/test_cholesky.py::TestCholesky_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_1__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never_param_3_{dtype=float64, shape=(1, 1)}::test_double_backward`
https://jenkins.preferred.jp/job/chainer/job/cupy_pr/597/TEST=chainer-py35,label=mn1-p100/console

<details>

```
chainer.testing.function_link.FunctionTestError: Parameterized test failed.

Base test method: TestCholesky_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_1__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never.test_double_backward
Test parameters:
  dtype: <class 'numpy.float64'>
  shape: (1, 1)


(caused by)
FunctionTestError: double backward is not implemented correctly

(caused by)
AssertionError: check_double_backward failed (eps=0.001 atol=0.001 rtol=0.001)
input[0]:
[[ 0.01938454]]
grad_output[0]:
[[-0.10458099]]
grad_grad_input[0]:
[[-0.56032458]]

check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
inputs[0]:
[[ 0.01938454]]
inputs[1]:
[[-0.10458099]]
grad_outputs[0]:
[[-0.56032458]]
directions[0]:
[[-0.95674023]]
directions[1]:
[[-0.29094352]]
gradients (numeric):  5.787207901617026
gradients (backward): 5.778750144208023

x: numeric gradient, y: backward gradient
Not equal to tolerance rtol=0.001, atol=0.001

(mismatch 100.0%)
 x: array(5.787207901617026)
 y: array(5.778750144208023)

assert_allclose failed: 
  shape: () ()
  dtype: float64 float64
  i: (0,)
  x[i]: 5.787207901617026
  y[i]: 5.778750144208023
  relative error[i]: 0.001463596313725404
  absolute error[i]: 0.008457757409003008
  relative tolerance * |y[i]|: 0.005778750144208023
  absolute tolerance: 0.001
  total tolerance: 0.006778750144208023
x: 5.787207901617026
y: 5.778750144208023


(caused by)
AssertionError: check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
inputs[0]:
[[ 0.01938454]]
inputs[1]:
[[-0.10458099]]
grad_outputs[0]:
[[-0.56032458]]
directions[0]:
[[-0.95674023]]
directions[1]:
[[-0.29094352]]
gradients (numeric):  5.787207901617026
gradients (backward): 5.778750144208023

x: numeric gradient, y: backward gradient
Not equal to tolerance rtol=0.001, atol=0.001

(mismatch 100.0%)
 x: array(5.787207901617026)
 y: array(5.778750144208023)

assert_allclose failed: 
  shape: () ()
  dtype: float64 float64
  i: (0,)
  x[i]: 5.787207901617026
  y[i]: 5.778750144208023
  relative error[i]: 0.001463596313725404
  absolute error[i]: 0.008457757409003008
  relative tolerance * |y[i]|: 0.005778750144208023
  absolute tolerance: 0.001
  total tolerance: 0.006778750144208023
x: 5.787207901617026
y: 5.778750144208023
```

<details>

<summary>Another example</summary>

```
chainer.testing.function_link.FunctionTestError: Parameterized test failed.

Base test method: TestCholesky_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_0__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never.test_double_backward
Test parameters:
  dtype: <class 'numpy.float64'>
  shape: (5, 5)


(caused by)
FunctionTestError: double backward is not implemented correctly

(caused by)
AssertionError: check_double_backward failed (eps=0.001 atol=0.001 rtol=0.001)
input[0]:
[[ 0.34759828  0.03582943 -0.13143365 -0.25388754  0.05630665]
 [ 0.03582943  0.31312966  0.05677956 -0.11083358 -0.29697093]
 [-0.13143365  0.05677956  0.20085346  0.07373341  0.03330158]
 [-0.25388754 -0.11083358  0.07373341  0.36120607 -0.05181148]
 [ 0.05630665 -0.29697093  0.03330158 -0.05181148  0.52502288]]
grad_output[0]:
[[ 0.62960863 -0.63435243 -0.02425354 -0.31533163 -0.28765864]
 [-0.03061119  0.60112378  0.64147257 -0.47950619 -0.70542231]
 [-0.3553684  -0.33798689 -0.37207768  0.98672684 -0.14900746]
 [ 0.56796335  0.8516327  -0.30540054  0.50900436  0.02241335]
 [-0.28623948  0.33110997  0.74031838  0.80731259 -0.56236024]]
grad_grad_input[0]:
[[ 0.43757224  0.72357597  0.47701812  0.66052396 -0.93172886]
 [-0.84203934  0.20341769 -0.13949926  0.5385612  -0.83432468]
 [ 0.88001246 -0.50311849  0.05465599 -0.82741954 -0.76840419]
 [ 0.40652236  0.39578807 -0.6596125  -0.57916194 -0.77416456]
 [ 0.26208706 -0.24078086  0.9625374  -0.65568732  0.92775172]]

check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
inputs[0]:
[[ 0.34759828  0.03582943 -0.13143365 -0.25388754  0.05630665]
 [ 0.03582943  0.31312966  0.05677956 -0.11083358 -0.29697093]
 [-0.13143365  0.05677956  0.20085346  0.07373341  0.03330158]
 [-0.25388754 -0.11083358  0.07373341  0.36120607 -0.05181148]
 [ 0.05630665 -0.29697093  0.03330158 -0.05181148  0.52502288]]
inputs[1]:
[[ 0.62960863 -0.63435243 -0.02425354 -0.31533163 -0.28765864]
 [-0.03061119  0.60112378  0.64147257 -0.47950619 -0.70542231]
 [-0.3553684  -0.33798689 -0.37207768  0.98672684 -0.14900746]
 [ 0.56796335  0.8516327  -0.30540054  0.50900436  0.02241335]
 [-0.28623948  0.33110997  0.74031838  0.80731259 -0.56236024]]
grad_outputs[0]:
[[ 0.43757224  0.72357597  0.47701812  0.66052396 -0.93172886]
 [-0.84203934  0.20341769 -0.13949926  0.5385612  -0.83432468]
 [ 0.88001246 -0.50311849  0.05465599 -0.82741954 -0.76840419]
 [ 0.40652236  0.39578807 -0.6596125  -0.57916194 -0.77416456]
 [ 0.26208706 -0.24078086  0.9625374  -0.65568732  0.92775172]]
directions[0]:
[[-0.07186171  0.11280323 -0.2742853  -0.05376069 -0.36955097]
 [ 0.07515146  0.12891765  0.01418645  0.15106081 -0.08371773]
 [-0.13889254 -0.0850225   0.05622033 -0.06729222 -0.22824911]
 [-0.23574125  0.09890549 -0.05435938 -0.11185858  0.09634402]
 [-0.158822    0.30513796 -0.10330198  0.21092425  0.08639563]]
directions[1]:
[[-0.09067345  0.20059894 -0.26440101 -0.05768598  0.04122594]
 [-0.12082885  0.089733    0.15899935 -0.07850528  0.02058059]
 [ 0.11824868  0.05286838  0.02654485  0.13107933 -0.09012302]
 [ 0.04476912 -0.04399981  0.06195626 -0.025265   -0.2622178 ]
 [-0.02999978 -0.12859212  0.05255747  0.17463968  0.09580609]]
gradients (numeric):  13.515421420773956
gradients (backward): 13.500351338334376

x: numeric gradient, y: backward gradient
Not equal to tolerance rtol=0.001, atol=0.001

Mismatch: 100%
Max absolute difference: 0.01507008
Max relative difference: 0.00111627
 x: array(13.515421)
 y: array(13.500351)

assert_allclose failed: 
  shape: () ()
  dtype: float64 float64
  i: (0,)
  x[i]: 13.515421420773956
  y[i]: 13.500351338334376
  relative error[i]: 0.0011162733518488921
  absolute error[i]: 0.01507008243958019
  relative tolerance * |y[i]|: 0.013500351338334377
  absolute tolerance: 0.001
  total tolerance: 0.014500351338334376
x: 13.51542142
y: 13.50035134


(caused by)
AssertionError: check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
inputs[0]:
[[ 0.34759828  0.03582943 -0.13143365 -0.25388754  0.05630665]
 [ 0.03582943  0.31312966  0.05677956 -0.11083358 -0.29697093]
 [-0.13143365  0.05677956  0.20085346  0.07373341  0.03330158]
 [-0.25388754 -0.11083358  0.07373341  0.36120607 -0.05181148]
 [ 0.05630665 -0.29697093  0.03330158 -0.05181148  0.52502288]]
inputs[1]:
[[ 0.62960863 -0.63435243 -0.02425354 -0.31533163 -0.28765864]
 [-0.03061119  0.60112378  0.64147257 -0.47950619 -0.70542231]
 [-0.3553684  -0.33798689 -0.37207768  0.98672684 -0.14900746]
 [ 0.56796335  0.8516327  -0.30540054  0.50900436  0.02241335]
 [-0.28623948  0.33110997  0.74031838  0.80731259 -0.56236024]]
grad_outputs[0]:
[[ 0.43757224  0.72357597  0.47701812  0.66052396 -0.93172886]
 [-0.84203934  0.20341769 -0.13949926  0.5385612  -0.83432468]
 [ 0.88001246 -0.50311849  0.05465599 -0.82741954 -0.76840419]
 [ 0.40652236  0.39578807 -0.6596125  -0.57916194 -0.77416456]
 [ 0.26208706 -0.24078086  0.9625374  -0.65568732  0.92775172]]
directions[0]:
[[-0.07186171  0.11280323 -0.2742853  -0.05376069 -0.36955097]
 [ 0.07515146  0.12891765  0.01418645  0.15106081 -0.08371773]
 [-0.13889254 -0.0850225   0.05622033 -0.06729222 -0.22824911]
 [-0.23574125  0.09890549 -0.05435938 -0.11185858  0.09634402]
 [-0.158822    0.30513796 -0.10330198  0.21092425  0.08639563]]
directions[1]:
[[-0.09067345  0.20059894 -0.26440101 -0.05768598  0.04122594]
 [-0.12082885  0.089733    0.15899935 -0.07850528  0.02058059]
 [ 0.11824868  0.05286838  0.02654485  0.13107933 -0.09012302]
 [ 0.04476912 -0.04399981  0.06195626 -0.025265   -0.2622178 ]
 [-0.02999978 -0.12859212  0.05255747  0.17463968  0.09580609]]
gradients (numeric):  13.515421420773956
gradients (backward): 13.500351338334376

x: numeric gradient, y: backward gradient
Not equal to tolerance rtol=0.001, atol=0.001

Mismatch: 100%
Max absolute difference: 0.01507008
Max relative difference: 0.00111627
 x: array(13.515421)
 y: array(13.500351)

assert_allclose failed: 
  shape: () ()
  dtype: float64 float64
  i: (0,)
  x[i]: 13.515421420773956
  y[i]: 13.500351338334376
  relative error[i]: 0.0011162733518488921
  absolute error[i]: 0.01507008243958019
  relative tolerance * |y[i]|: 0.013500351338334377
  absolute tolerance: 0.001
  total tolerance: 0.014500351338334376
x: 13.51542142
y: 13.50035134
```

--- a/tests/chainer_tests/functions_tests/math_tests/test_cholesky.py
+++ b/tests/chainer_tests/functions_tests/math_tests/test_cholesky.py
@@ -56,8 +56,9 @@ class TestCholesky(testing.FunctionTestCase):
 
     def setUp(self):
         self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-3}
-        self.check_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
-        self.check_double_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
+        self.check_backward_options = {'atol': 1e-3, 'rtol': 1e-3, 'eps': 1e-4}
+        self.check_double_backward_options = {
+            'atol': 1e-3, 'rtol': 1e-3, 'eps': 1e-4}
 
     def generate_inputs(self):
         a = self.random_matrix(self.shape, self.dtype, scale=(1e-2, 2.0),

